xio-core/src/test/java/com/xjeffrose/xio/http/GrpcFunctionalTest.java

```diff
@@ -375,15 +379,22 @@ public Client createClient(ChannelHandlerContext ctx, ClientConfig config) {
             return new Client(clientState, () -> new ProxyBackendHandler(ctx));
           }
         };
+
+    ApplicationState appState =
+        new ApplicationState(ApplicationConfig.fromConfig("xio.defaultApplication"));
+    XioServerConfig serverConfig = XioServerConfig.fromConfig("xio.testGrpcServer");
+
     XioServerBootstrap bootstrap =
-        XioServerBootstrap.fromConfig("xio.testGrpcServer")
+        new XioServerBootstrap(appState, serverConfig, new XioServerState(serverConfig))
```

Comment:
White spacing issues. Also, a comment to describe what you're doing here would help. Lots of chain magic happening here.

Reply:
We should document a rule of thumb about number of chains allowed before comments should be added.

Reply:
Agreed.